### ISSUE TITLE
New tsi1 index and missing datapoints

### ISSUE THREAD

**Issue by davemt** [influxdata/influxdb#8571]
When new `tsi1` index is enabled, some datapoints submitted via HTTP POST go missing, despite being logged when http write-tracing is turned on. In our case these are submitted from the `Heapster` service on our Kubernetes cluster with influxdb set as `--sink`.

__System info:__

- InfluxDB version 1.3.0
- Running as container, official influxdb:1.3 docker image, in kubernetes AWS w/ mounted ext4 EBS volume
- `tsi1` index enabled

__Steps to reproduce:__

1. Configure influxdb to use `index-version = tsi1` and start `heapster` service (with no database existing yet)
2. Wait for create db and pushes from `heapster`
3. Query for expected datapoints (e.g., `select * from "k8s"."default"."cpu/usage_rate" where type = 'node' limit 1`).

__Expected behavior:__

Expect that any datapoint that is submitted via HTTP POST and logged via http write-tracing would make it into the measurement and be query-able.

__Actual behavior:__

Some (but not all) datapoints are missing (or un-query-able). _When new index is turned off in favor of default original and repro steps are run, this problem goes away._

__Additional info:__

```
> use k8s
> select * from "cpu/usage_rate" where type = 'node' limit 1
> 
```
From influxdb log (http write-tracing enabled; IPs X'd out). Note `type=node` tag:
```
cpu/usage_rate,cluster_name=default,labels=beta.kubernetes.io/arch:amd64\,beta.kubernetes.io/instance-type:m4.10xlarge\,beta.kubernetes.io/os:linux\,failure-domain.beta.kubernetes.io/region:us-east-2\,failure-domain.beta.kubernetes.io/zone:us-east-2b\,isolation:loadgen\,kubernetes.io/hostname:ip-X-X-X-X.us-east-2.compute.internal\,kubernetes.io/role:node\,node-role.kubernetes.io/node:,nodename=ip-X-X-X-X.us-east-2.compute.internal,type=node value=15307i 1499352540000000000
```

I see no indication of errors in influx log (or on client side for that matter). HTTP POSTs are OK responses. I turned trace logging for `[data]` section on as well.

**Comment by jwilder** [comment]
@benbjohnson Would #8539 fix this?  Looks like there is some escaping in the tags.

**Comment by benbjohnson** [comment]
@jwilder Yes, #8539 would likely fix the escaping issues on read. It should be saved to the index correctly.

**Comment by davemt** [comment]
I noticed changelog did not include information about release of the fix, #8539. Is it expected it will make it into 1.3.1 release (and if so is there some rough idea of when this will be out)?

**Comment by rbetts** [comment]
@davemt https://github.com/influxdata/influxdb/pull/8540/ merges the fix for this issue to the 1.3 GA release (1.3.1).
